// @mescius/spread-sheets-angular/esm2020/src/spreadSheets.module.mjs
/*import_begin*/
import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';
import { ColumnComponent } from './column.component';
/*import_end*/
import { SpreadSheetsComponent } from './spreadSheets.component';
import { WorksheetComponent } from './worksheet.component';
import * as i0 from "@angular/core";
/*code_begin*/
export class SpreadSheetsModule {
}
SpreadSheetsModule.ɵfac = i0.ɵɵngDeclareFactory({ minVersion: "12.0.0", version: "13.4.0", ngImport: i0, type: SpreadSheetsModule, deps: [], target: i0.ɵɵFactoryTarget.NgModule });
SpreadSheetsModule.ɵmod = i0.ɵɵngDeclareNgModule({ minVersion: "12.0.0", version: "13.4.0", ngImport: i0, type: SpreadSheetsModule, declarations: [SpreadSheetsComponent, WorksheetComponent, ColumnComponent], imports: [CommonModule], exports: [SpreadSheetsComponent, WorksheetComponent, ColumnComponent] });
SpreadSheetsModule.ɵinj = i0.ɵɵngDeclareInjector({ minVersion: "12.0.0", version: "13.4.0", ngImport: i0, type: SpreadSheetsModule, imports: [[CommonModule]] });
i0.ɵɵngDeclareClassMetadata({ minVersion: "12.0.0", version: "13.4.0", ngImport: i0, type: SpreadSheetsModule, decorators: [{
            type: NgModule,
            args: [{
                    imports: [CommonModule],
                    declarations: [SpreadSheetsComponent, WorksheetComponent, ColumnComponent],
                    exports: [SpreadSheetsComponent, WorksheetComponent, ColumnComponent],
                }]
        }] });
